dfdaemon/server.go

```diff
@@ -117,7 +117,9 @@ func NewFromConfig(cfg config.Properties) (*Server, error) {
 
 // Start runs dfdaemon's http server.
 func (s *Server) Start() error {
-	proxy.WithDirectHandler(handler.New())(s.proxy)
+	if err := proxy.WithDirectHandler(handler.New())(s.proxy); err != nil {
```

Comment:
[proxy.WithDirectHandler](https://github.com/dragonflyoss/Dragonfly/blob/master/dfdaemon/proxy/proxy.go#L81) always returns nil. Maybe we can just leave it be? Or explicitly use

```
_ = proxy.WithDirectHandler(handler.New())(s.proxy)
```

to show that we've ignored the error.

Reply:
Thank you!
you are right,I will fix it.
I think proxy.WithDirectHandler always return nil is not good.should we fix it?

Reply:
I am on `_ = proxy.WithDirectHandler(handler.New())(s.proxy)`. 
I agree with that we usually do not ignore returned instance when we can, but `_` could make it more clear.

Reply:
> I think proxy.WithDirectHandler always return nil is not good.should we fix it?

The reason it returns an error is that `proxy.WithDirectHandler` is meant to be used as `proxy.Option`, which is `func (p *proxy.Proxy) error`. The nil error is only used to satisfy the function signature.